src/js_realm.hpp

```diff
@@ -1382,41 +1382,151 @@ void RealmClass<T>::compact(ContextType ctx, ObjectType this_object, Arguments&
+ * @param return_value none
+ */
 template <typename T>
 void RealmClass<T>::writeCopyTo(ContextType ctx, ObjectType this_object, Arguments& args, ReturnValue& return_value)
 {
     args.validate_maximum(2);
 
+    enum CopyMode {
+        COPYMODE_WITHPATH = 1,
+        COPYMODE_WITHCONFIG = 1 << 1,
+        COPYMODE_TOSYNC = 1 << 2,
+        COPYMODE_TOLOCAL = 1 << 3,
+        COPYMODE_ENCRYPTED = 1 << 4,
+    };
+
+    size_t copy_mode = 0;
+
     if (args.count == 0) {
-        throw std::runtime_error("At least path has to be provided for 'writeCopyTo'");
+        throw std::runtime_error("`writeCopyTo` requires an output configuration or path");
     }
 
     SharedRealm realm = *get_internal<T, RealmClass<T>>(ctx, this_object);
 
-    ValueType pathValue = args[0];
-    if (!Value::is_string(ctx, pathValue)) {
-        throw std::runtime_error("Argument to 'writeCopyTo' must be a String.");
+    ValueType firstParamValue = args[0];
+    if (Value::is_object(ctx, firstParamValue)) {
+        copy_mode = COPYMODE_WITHCONFIG;
+    }
+    else if (Value::is_string(ctx, firstParamValue)) {
+        copy_mode = COPYMODE_WITHPATH;
+    }
+    else {
+        throw std::runtime_error("First argument to 'writeCopyTo' must be a config Object or an output path");
     }
 
-    std::string path = Value::validated_to_string(ctx, pathValue);
+    std::string output_path;
+    OwnedBinaryData encryption_key;
+    ObjectType output_config;
 
-    if (args.count == 1) {
-        BinaryData empty_encryption_key;
-        realm->write_copy(path, empty_encryption_key);
+    // switch on the first two bits of copy_mode to parse overload of `writeCopyTo()` function from JS
+    switch (copy_mode & 3) {
+        case COPYMODE_WITHCONFIG: {
+            if (args.count > 1) {
+                throw std::runtime_error("`writeCopyTo(<config>)` accepts only one parameter");
+            }
 
-        return;
-    }
+            output_config = Value::validated_to_object(ctx, firstParamValue);
 
-    // enryption key is specified
-    ValueType encryption_key_arg = args[1];
+            // make sure that `path` property exists and that it is a string
+            ValueType pathValue = Object::get_property(ctx, output_config, "path");
+            if (Value::is_string(ctx, pathValue)) {
+                output_path = Value::to_string(ctx, pathValue);
+            }
+            else if (!Value::is_valid(pathValue)) {
+                throw std::runtime_error("`path` property must exist in output configuration");
+            }
+            else {
+                throw std::runtime_error("`path` property must be a string");
+            }
+
+            // check whether encryption key property exists, and whether it's a string
+            ValueType encKeyValue = Object::get_property(ctx, output_config, "encryptionKey");
+            // `encryptionKey` is optional..
+            if (Value::is_valid(encKeyValue)) {
```

Comment:
does `is_valid` return true if it's not set?

Reply:
`is_valid` returns the equivalent of
`return _value == nullptr;`
I.e., it tells you whether the `Value`'s internal data pointer points to something that isn't `null`